### PR TITLE
Feature/kingpin

### PR DESCRIPTION
**Pull request by bradcooper314**
Oh boy is the physics a mess. It's not the best design, probably not even a great design, but it's a working and functional design. Will need to revamp a lot of the physics system if we want to make projectiles great again.

Also, it's hilarious watching them fly off the ramp and into the nothingness. Maybe an air resistance value or something, but I've already spent nearly 5 hours on something that I didn't account for.

### CODE REVIEW

File: CS483/CS483/Kartaclysm/Components/Abilities/ComponentStrikeAbility.cpp [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		{
+			ParseNode(p_pBaseNode, strProjectileXML);
+		}
+		if (p_pOverrideNode != nullptr)
+		{
+			ParseNode(p_pOverrideNode, strProjectileXML);
+		}
+
+		// Check that we got everything we needed.
+		assert(strProjectileXML != "");
```
Comment (MatthewWhite) [inline CS483/CS483/Kartaclysm/Components/Abilities/ComponentStrikeAbility.cpp:55]: Again you're saying it's optional, then asserting it's not the default.  Which is it?

Comment (MatthewWhite) [reply CS483/CS483/Kartaclysm/Components/Abilities/ComponentStrikeAbility.cpp:55]: well now I look like quite the fool

File: CS483/CS483/Kartaclysm/Components/Abilities/ComponentTurkeyAbility.cpp [hunk 2]
```diff
@@ -0,0 +1,108 @@
+//----------------------------------------------------------------------------
+// ComponentTurkeyAbility.h
+// Author: Bradley Cooper
+//
+// Speedster's ability to give a temporary speed boost.
```
Comment (MatthewWhite) [inline CS483/CS483/Kartaclysm/Components/Abilities/ComponentTurkeyAbility.cpp]: Kingpin

File: CS483/CS483/Kartaclysm/Components/Abilities/ComponentTurkeyAbility.h [hunk 3]
```diff
@@ -0,0 +1,74 @@
+//----------------------------------------------------------------------------
+// ComponentTurkeyAbility.h
+// Author: Bradley Cooper
+//
+// Speedster's ability to give a temporary speed boost.
```
Comment (MatthewWhite) [inline CS483/CS483/Kartaclysm/Components/Abilities/ComponentTurkeyAbility.h]: Same as ComponentTurkeyAbility.cpp

File: CS483/CS483/Kartaclysm/Components/ComponentProjectile.h [hunk 4]
```diff
@@ -0,0 +1,73 @@
+//----------------------------------------------------------------------------
+// ComponentProjectile.cpp
```
Comment (MatthewWhite) [inline CS483/CS483/Kartaclysm/Components/ComponentProjectile.h]: ComponentProjectile.h

File: CS483/CS483/Kartaclysm/Components/ComponentSelfDestruct.h [hunk 5]
```diff
@@ -0,0 +1,68 @@
+//----------------------------------------------------------------------------
+// ComponentSelfDestruct.cpp
```
Comment (MatthewWhite) [inline CS483/CS483/Kartaclysm/Components/ComponentSelfDestruct.h]: Oh my god don't just blindly copy and paste!

File: CS483/CS483/Kartaclysm/Data/kart.xml.orig [hunk 6]
```diff
@@ -0,0 +1,32 @@
+<!-- Kart -->
```
Comment (MatthewWhite) [inline CS483/CS483/Kartaclysm/Data/kart.xml.orig]: Why has this file been pushed?

**Review by MatthewWhite — CHANGES_REQUESTED** [review]
A few minor things to change, then I'll approve it

File: CS483/CS483/Kartaclysm/Components/ComponentSelfDestruct.h (hunk 5 above)
Comment (bradcooper314) [reply CS483/CS483/Kartaclysm/Components/ComponentSelfDestruct.h]: Is it that obvious?

**Review by MatthewWhite — APPROVED** [review]

**Pull request merged**